internal/agent/install.go

```diff
@@ -289,8 +294,9 @@ func dumpCCStringToFile(c *config.Config) (string, error) {
 		c.Logger.Error("Error creating temporary file for install config: %s\n", err.Error())
 		return "", err
 	}
-	defer os.RemoveAll(f.Name())
-
+	defer func(f *os.File) {
+		_ = f.Close()
```

Comment:
no cleanup?

Reply:
closing the file here is correct. Removing it should be the responsibility of the caller. The temp directory is gone after installation right? Leaving it there is ok then.